File: java/busyschedule.java
import java.util.Arrays;
import java.util.Scanner;

public class busyschedule {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int n;
		while ((n = scan.nextInt()) != 0) {
			scan.nextLine();
			Time[] schedule = new Time[n];
			for(int i = 0; i < n; i++)
				schedule[i] = new Time(scan.nextLine());
			Arrays.sort(schedule);
			for(int i = 0; i < n; i++)
				System.out.println(schedule[i]);
			System.out.println();
		}
		scan.close();
	}
}

class Time implements Comparable<Time> {
	int hour;
	int minute;
	String meridiem;
	
	public Time(String t) {
		String[] t_1 = t.split(" ");
		meridiem = t_1[1];
		String[] t_2 = t_1[0].split(":");
		hour = Integer.parseInt(t_2[0]);
		minute = Integer.parseInt(t_2[1]);
	}

	public int compareTo(Time other) {
		if (!meridiem.equals(other.meridiem))
			return meridiem.compareTo(other.meridiem);
		else if (hour != other.hour)
			if (hour != 12 && other.hour != 12)
				return hour - other.hour;
			else {
				if (hour == 12)
					return -other.hour;
				else
					return hour;
			}
		else
			return minute - other.minute;
	}
	
	public String toString() {
		return hour + ":" + ("00" + minute).substring(String.valueOf(minute).length()) + " " + meridiem;
	}
}

File: ts/src/busyschedule.ts
import readline = require('readline')

class Schedule {
  hour: number
  minute: number
  meridiem: string

  constructor (time: string) {
    const timeArr = time.split(' ')
    this.meridiem = timeArr[1] ?? ''
    ;[this.hour, this.minute] = (timeArr[0] ?? '').split(':').map(Number) as [
      number,
      number
    ]
  }

  public toString (): string {
    return `${this.hour}:${this.minute.toString().padStart(2, '0')} ${
      this.meridiem
    }`
  }

  static sort (a: Schedule, b: Schedule): number {
    if (a.meridiem !== b.meridiem) {
      return a.meridiem.localeCompare(b.meridiem)
    }
    if (a.hour === b.hour) {
      return a.minute - b.minute
    }
    if (a.hour === 12) {
      return -1
    }
    if (b.hour === 12) {
      return 1
    }
    return a.hour - b.hour
  }
}

const digitsOnly = /^\d+$/
let schedules: Schedule[] = []

readline
  .createInterface({
    input: process.stdin,
    output: process.stdout
  })
  .on('line', (line: string) => {
    if (digitsOnly.test(line)) {
      if (schedules.length > 0) {
        for (const schedule of schedules.sort(Schedule.sort)) {
          console.log(schedule.toString())
        }
        console.log()
      }
      const schedulesCount = parseInt(line)
      if (schedulesCount === 0) {
        process.exit(0)
      }
      schedules = []
    } else {
      schedules.push(new Schedule(line))
    }
  })
  .on('close', () => {
    process.exit(0)
  })
